### PR TITLE
Repair illegal substitutions to an inversion

### PR DESCRIPTION
**Pull request by nuggetoriniku**
I added two repairer functions to correct illegal hgvs strings.
Could you review this?

### CODE REVIEW

File: src/clj_hgvs/repairer.cljc [hunk 1]
```diff
@@ -102,6 +108,19 @@
     (string/replace s #"(del|dup|inv)\d+$" "$1")
     s))
 
+;; c.2210_2211CA>TG	-> c.2210_2211inv
+(defn ^:no-doc basis->inv
+  [s kind]
+  (let [base-pairs {\A \T \T \A \C \G \G \C}
```
Comment (totakke) [inline src/clj_hgvs/repairer.cljc:114]: An inversion mutation is not appeared in protein mutations.

```clj
(if (#{:genome :mitochondria :coding-dna :non-coding-dna :circular-dna :rna}
     kind)
  ...
  s)
```

File: src/clj_hgvs/repairer.cljc [hunk 2]
```diff
@@ -102,6 +108,19 @@
     (string/replace s #"(del|dup|inv)\d+$" "$1")
     s))
 
+;; c.2210_2211CA>TG	-> c.2210_2211inv
+(defn ^:no-doc basis->inv
```
Comment (totakke) [inline src/clj_hgvs/repairer.cljc:112]: Rename to `substitutions->inv` and add it to above `substitution->delins` in [`built-in-repairers`](https://github.com/chrovis/clj-hgvs/blob/a9203133f219c7fbfc1b0fb0962bc3be520f63bf/src/clj_hgvs/repairer.cljc#L286).

File: src/clj_hgvs/repairer.cljc [hunk 3]
```diff
@@ -102,6 +108,19 @@
...
+  [s kind]
+  (let [base-pairs {\A \T \T \A \C \G \G \C}
+        [_ region b a] (re-find #"(\d+_\d+)([A-Z]+)>([A-Z]+)" s)]
+    (if (= a
+           (->> b
+                (map base-pairs)
+                reverse
+                (apply str)))
+      (str "c." region "inv")
```
Comment (totakke) [inline src/clj_hgvs/repairer.cljc:121]: The regex does not cover all coordinate patterns. Additionally, I recommend using `replace` fn.

```clj
(letfn [(revcomp [s]
          (->> (reverse s)
               (map {\A \T \T \A \C \G \G \C})
               (apply str)))]
  (string/replace s
                  #"([-\d+*]+_[-\d+*]+)([A-Z]{2,})>([A-Z]{2,})"
                  (fn [[s coords del ins]]
                    (if (= del (revcomp ins))
                      (str coords "inv")
                      s))))
```

**Review by totakke — CHANGES_REQUESTED** [review]
Thank you for PR. I added some comments. And you should add test cases to [`clj-hgvs.core-test/repair-hgvs-str-test`](https://github.com/chrovis/clj-hgvs/blob/a9203133f219c7fbfc1b0fb0962bc3be520f63bf/test/clj_hgvs/core_test.cljc#L251).

File: src/clj_hgvs/repairer.cljc [hunk 4]
```diff
@@ -87,6 +87,12 @@
     (string/replace s #"([-\d+*]+_[-\d+*]+)([A-Z]+)?>([A-Z]+)" "$1del$2ins$3")
     s))
 
+;; c.233_234TC>CT	-> c.233_234delinsCT
+(defn ^:no-doc substitutions->delins*
```
[the remark this answers is not in the document]
Comment (nuggetoriniku) [reply src/clj_hgvs/repairer.cljc]: Thank you for your review.
I removed `substitutions->delins*` function.

File: src/clj_hgvs/repairer.cljc (hunk 1 above)
Comment (nuggetoriniku) [reply src/clj_hgvs/repairer.cljc]: I set the condition for exclude `:protein`

File: src/clj_hgvs/repairer.cljc (hunk 2 above)
Comment (nuggetoriniku) [reply src/clj_hgvs/repairer.cljc]: I renamed to `substitutions->inv` and added it in `built-in-repairers`.

File: src/clj_hgvs/repairer.cljc (hunk 3 above)
Comment (nuggetoriniku) [reply src/clj_hgvs/repairer.cljc]: I fixed the regex to cover all coordinate patterns, and changed process with using `replace` fn.

**Review by totakke — CHANGES_REQUESTED** [review]
Thanks, your fix is perfect, but a test is still missing (https://github.com/chrovis/clj-hgvs/pull/3#pullrequestreview-477303755).

Could you add `c.2210_2211CA>TG` -> `c.2210_2211inv` test case to just above the following line? https://github.com/chrovis/clj-hgvs/blob/a9203133f219c7fbfc1b0fb0962bc3be520f63bf/test/clj_hgvs/core_test.cljc#L286

**Review by totakke — APPROVED** [review]
LGTM. I plan to merge it after @federkasten approval.

**Review by federkasten — APPROVED** [review]
Sorry for the delay in reviewing.
I have checked. No additional comments. Thanks! :+1:

**Pull request merged**